text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+When considering artifact provenance, one of the long known weaknesses in the OSS world are the central official package management systems that serve as a nexus for sharing and consumption of artifacts in any given language ecosystem. Historically these systems have struggled with a variety of attack vectors and suffered most specifically from a lack of concrete cryptographic artifact identities and security focused policies and tooling to manage contribution and ongoing artifact maintenance. The challenges are further exacerbated by the state of build system fidelity broadly. Provenance generation at artifact creation time and subsequent verification is where Sigstore comes in. Its approach, leveraging OIDC based identities as a gate to the signing function as opposed to long-term sensitive keys, as well as a public root and widely available verification capabilities make it a developer friendly, potentially ubiquitous signing architecture.
+
+This document will address two primary use case scenarios:
+
+1. The developer building Crates locally on their desktop
+1. Crates produced through a formal build plant (GitHub Actions, CircleCI, etc. and commercial enterprise internal build plants)
+
+These use cases will allow us to start to draw trust boundaries around artifact production processes and identify the metadata attributes that will need to be generated at artifact build-time which are subsequently consumed at run-time for effective policy and consumption controls to be possible for the ultimate consumers.
+
+It’s worth noting that the team at GitHub working on the NPM implementation have essentially landed at the same to use case patterns. In our discussions with them as this document was prepared they articulated a focus on the build plant pattern as that pattern allows for the establishment of an attestation story around the systems where the artifact was produced. Conversely, artifacts produced on a developer desktop or system that is not formally controlled from a security perspective must necessarily be classified as technically “unsafe”.
```

Comment:
Is this actually relevant to crates.io?  After all, it usually distributes only source code, not compiled artifacts.  I suppose people sometimes stick compiled packages into crates.io "source" packages, e.g. I think `watt` is based on this approach.  But that doesn't seem important enough to mention in the summary.